analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Binary.cs

```diff
@@ -531,10 +532,10 @@ public void BinaryNotEqualsNull_SetsBoolConstraint_KnownResult_VB()
 Tag(""EqualsFalse"", EqualsFalse)";
         var validator = SETestContext.CreateVB(code).Validator;
         validator.ValidateContainsOperation(OperationKind.Binary);
-        validator.ValidateTag("IsTrue", x => x.HasConstraint(BoolConstraint.True).Should().BeTrue());
-        validator.ValidateTag("IsFalse", x => x.HasConstraint(BoolConstraint.False).Should().BeTrue());
-        validator.ValidateTag("EqualsTrue", x => x.HasConstraint(BoolConstraint.True).Should().BeTrue());
-        validator.ValidateTag("EqualsFalse", x => x.HasConstraint(BoolConstraint.False).Should().BeTrue());
+        validator.TagValue("IsTrue").Should().HaveOnlyConstraints(BoolConstraint.True, ObjectConstraint.NotNull);
+        validator.TagValue("IsFalse").Should().HaveOnlyConstraints(BoolConstraint.False, ObjectConstraint.NotNull);
+        validator.TagValue("EqualsTrue").Should().HaveOnlyConstraints(BoolConstraint.True); // TODO ObjectConstraint.NotNull missing
```

Comment:
Do we need an issue for this? @pavel-mikula-sonarsource

Reply:
This UT is actually wrong. `<>` operator is not defined for object and null. This compiles only because of `option strict off` (stupid default configuration).

I would delete all lines that have `EqualsTrue` and `EqualsFalse` from this UT.

Reply:
I was baffled by `something <> Nothing` and I was wondering why this even compiles. So thanks for the explanation. (This reminds me of the even worse [Option Explicit](https://learn.microsoft.com/en-us/office/vba/language/reference/user-interface-help/option-explicit-statement) in VBA (also off by default).

Reply:
They took the wrong decision to have Option Explicit Off And Option Strict Off so people could migrate their flaky VB6 codebases :( They should have been both ON by default, and we have rules for that :)